deploy/local/vars_lnx.sh

```diff
@@ -15,77 +15,153 @@
+# see https://stackoverflow.com/a/28776166/2914328
+# see https://stackoverflow.com/a/60783610/2914328
+# see https://stackoverflow.com/a/2942183/2914328
+
+# This script is designed to be POSIX compatible, there are a few lines in the
+# code block below that contain elements that are specific to a shell. The
+# shell-specific elements are needed to identify the sourcing shell.
+
+vars_script_name="vars.sh"
+
+vars_sourced=0 ;
+vars_sourced_sh="$(ps -p "$$" -o  command= | awk '{print $1}')" ;
+vars_sourced_nm="$(ps -p "$$" -o  command= | awk '{print $2}')" ;
+
+# ${var:-} needed to pass "set -eu" checks
+# see https://unix.stackexchange.com/a/381465/103967
+# see https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02
+if [ -n "${ZSH_VERSION:-}" ] ; then     # only executed in zsh
+  vars_sourced=0 ; #echo "   ZSH version \"$ZSH_VERSION\""
+  vars_sourced_sh="zsh" ;               # only meaningful if vars_sourced=1
+  vars_sourced_nm="${(%):-%x}" ;        # ditto
+  if [ -n "$ZSH_EVAL_CONTEXT" ] ; then  # only present in zsh 5.x and later
+    case $ZSH_EVAL_CONTEXT in (*:file*) vars_sourced=1 ;; esac ;
+  fi
+elif [ -n "${KSH_VERSION:-}" ] ; then   # only executed in ksh or mksh or lksh
+  vars_sourced=0 ; #echo "   KSH version \"$KSH_VERSION\""
+  if [ "$(set | grep KSH_VERSION)" = "KSH_VERSION=.sh.version" ] ; then # ksh
+    if [ "$(cd "$(dirname -- "$0")" && pwd -P)/$(basename -- "$0")" \
+      != "$(cd "$(dirname -- "${.sh.file}")" && pwd -P)/$(basename -- "${.sh.file}")" ] ; then
+      vars_sourced=1 ;
+      vars_sourced_sh="ksh" ;           # only meaningful if sourced=1
+      vars_sourced_nm="${.sh.file}" ;   # ditto
+    fi
+  else # mksh or lksh detected (also check for [lm]ksh renamed as ksh)
+    vars_sourced_sh="$(basename -- "$0")"
+    if [ "mksh" = "$vars_sourced_sh" ] || [ "lksh" = "$vars_sourced_sh" ] || [ "ksh" = "$vars_sourced_sh" ] ; then
+      vars_sourced=1 ;
+      # force [lm]ksh to issue error msg; contains this script's rel/path/filename
+      vars_sourced_nm="$( (echo "${.sh.file}") 2>&1 )" || : ;
+      vars_sourced_nm="$(expr "$vars_sourced_nm" : '^.*ksh: \(.*\)\[[0-9]*\]:')" ;
+    fi
+  fi
+elif [ -n "${BASH_VERSION:-}" ] ; then  # only executed in bash
+  vars_sourced=0 ; #echo "   BASH version \"$BASH_VERSION\""
+  vars_sourced_sh="bash" ;              # only meaningful if vars_sourced=1
+  # shellcheck disable=2128
+  vars_sourced_nm="$BASH_SOURCE" ;      # ditto
+  (return 0 2>/dev/null) && vars_sourced=1 ;
+# TODO: following needs further testing to work in dash
+elif [ "${0:-}" = "dash" ] ; then
+  vars_sourced=1 ;                      # see error messages below for outcome
+  vars_sourced_sh="dash" ;              # see error messages below for outcome
+  vars_sourced_nm="$vars_sourced_nm" ;
+# Only reliable way to detect sh source is to know the name of this script.
+# TODO: following needs further testing to work in sh
```

Comment:
Have you checked now what works locally on Mac and Linux?

Reply:
dash would just fail

Reply:
actually i've just commit updated version with dash support